src/read_utils.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017-2018 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+use collection::CollectionWithId;
+use model::Collections;
+use objects::{self, Contributor};
+use std::collections::BTreeSet;
+use std::fs::File;
+use std::path;
+use utils::{add_prefix_to_collection, add_prefix_to_collection_with_id};
+use Result;
+extern crate serde_json;
+
+#[derive(Deserialize, Debug)]
+struct ConfigDataset {
+    dataset_id: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Config {
+    contributor: objects::Contributor,
+    dataset: ConfigDataset,
+}
+
+pub fn read_config<P: AsRef<path::Path>>(
+    config_path: Option<P>,
+) -> Result<(objects::Contributor, objects::Dataset)> {
+    let contributor;
+    let dataset;
+    if let Some(config_path) = config_path {
+        let json_config_file = File::open(config_path)?;
+        let config: Config = serde_json::from_reader(json_config_file)?;
+        info!("Reading dataset and contributor from config: {:?}", config);
+
+        contributor = config.contributor;
+
+        use chrono::{Duration, Utc};
+        let duration = Duration::days(15);
+        let today = Utc::today();
```

Comment:
Why?

Reply:
I think you forgotten to remove the old version of this function.

Reply:
you are right ! Using the new function implies some refacto, I've opened an issue for this point to separate issues : https://github.com/CanalTP/navitia_model/issues/98